src/resource.c

```diff
@@ -473,14 +466,22 @@ hsk_resource_has_ns(const hsk_resource_t *res) {
   return false;
 }
 
+void
+hsk_resource_write_synth(char *b32, uint8_t *name) {
```

Comment:
Here we can just use `memcpy` instead of sprintfs, sprintf does not make it more readable or easier to work with.

Reply:
Also it's more readable and consistent with other methods, if the `uint8_t *name` - where we write - is first and `char *b32` - we copy from - second arg.

Reply:
Here is the example of this method (not fully tested): https://github.com/nodech/hnsd-misc-c/tree/master/sprintf-v-memcpy

Reply:
> Also it's more readable and consistent with other methods, if the `uint8_t *name` - where we write - is first and `char *b32` - we copy from - second arg.

I didn't make this change yet because I think the hnsd style for these kinds of methods is more like `_write(in, out)`, such as `hsk_dns_msg_write(const hsk_dns_msg_t *msg, uint8_t **data)`. Even though I think you're right about standard methods like `memcpy(dest, src)` ? 

I copied your memcpy implementation of this function from your benchmark test, thanks! 

fixed in f5200ab1251950b897e8ef3f71f5a11737aeaa5a